aep/general/0193/aep.md.j2

```diff
@@ -1,5 +1,150 @@
+error occurs, and **must** use the canonical error codes defined in
+[`google.rpc.Code`][Code]. More information about the particular codes is
+available in the [gRPC status code documentation][].
+
+{% endtabs %}
+
+#### Dynamic variables
+
+The best, actionable error detail includes dynamic segments. These variable
+parts of the message are specific to a particular request. Consider the
+following example:
+
+> The Book, "The Great Gatsby", is unavailable at the Library, "Garfield East".
+> It is expected to be available again on 2199-05-13.
+
+The preceding error detail is made actionable by the context, both that
+originates from the request, the title of the Book and the name of the Library,
+and by the information that is known only by the service, i.e. the expected
+return date of the Book.
+
+All dynamic variables found in error detail **must** also be present as
+additional properties of the error response
+
+```yaml
+bookTitle: 'The Great Gatsby'
+library: 'Garfield East'
+expectedReturnDate: '2199-05-13'
+```
+
+Once present for a particular error type, additional properties **must**
+continue to be included in the the error response to be backwards compatible,
+even if the value for a particular property is empty.
+
+#### Localization
+
+The title and detail fields **should** be presented in the service's native
+language. If a localized detail is required, it may be included as an
+additional property in the error response, and **should** be named
+`localizedDetail`.
+
+### Partial errors
+
+APIs **should not** support partial errors. Partial errors add significant
+complexity for users, because they usually sidestep the use of error codes, or
+move those error codes into the response message, where the user must write
+specialized error handling logic to address the problem.
+
+However, occasionally partial errors are necessary, particularly in bulk
+operations where it would be hostile to users to fail an entire large request
+because of a problem with a single entry.
+
+Methods that require partial errors **should** use [long-running operations][],
+and the method **should** put partial failure information in the metadata
+message. The errors themselves **must** still be represented as an error object
+as described in this AEP.
+
+### Permission Denied
+
+If the user does not have permission to access the resource or parent,
+regardless of whether or not it exists, the service **must** error with
+`PERMISSION_DENIED` (HTTP 403). Permission **must** be checked prior to
+checking if the resource or parent exists.
+
+If the user does have proper permission, but the requested resource or parent
+does not exist, the service **must** error with `NOT_FOUND` (HTTP 404).
+
+## Rationale
+
+## Further reading
+
+- [RFC 9457 Problem Details for HTTP APIs][RFC 9457]
+- For which error codes to retry, see [AEP-194](https://aep.dev/194).
+
+## Changelog
+
+- **2024-03-21**: From from https://google.aip.dev/193
```

Comment:
```suggestion
- **2024-03-21**: Adopt from https://google.aip.dev/193
```

Fixes a typo, but also, I don't think we've been adding changelog messages like this in general - so maybe we leave it out?

Reply:
I thought we agreed to do this but happy to take it out if misunderstood. I did put this in AEP-8 and AEP-135 (which have the same typo), so we should delete from there if that's the direction we want to go.

Personally I think that having an entry for the initial version is a good thing, so I'll  fix the typo and leave it here until we have consensus on this. If we decide we don't want it then we can remove them all in a separate PR.

Reply:
SGTM